### PR TITLE
Fix: Attach instagram images with file type share

### LINKED ISSUE #4050: Investigation: Instagram DMs
Check if the Instagram DMs work with image/video previews.

### PR DESCRIPTION
**Pull request by tejaswinichile**
# Pull Request Template

## Description

Instagram image display for share and story mention.

Fixes #4050 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Tested on staging.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by tejaswinichile** [comment]
https://developers.facebook.com/docs/messenger-platform/instagram/features/webhook/ 

Followed and added this to handle all types of media that are supported by Facebook, like share and story_mention.

There is one limit though when sharing some videos mentioned in the story and we share that, Facebook sends message unsupported that we can't fetch or display in our conversation

```
 [{"time"=>1646241594500, "id"=>"17841449543715721", "messaging"=>[{"sender"=>{"id"=>"some_id"}, "recipient"=>{"id"=>"some_id"}, "timestamp"=>1646241593850, "message"=>{"mid"=>"some_mid", "is_unsupported"=>true}}]}]
```

These types of messages we can't handle, right now I have added the attachment handling for all supported type attachments.

**Comment by tejaswinichile** [comment]
https://user-images.githubusercontent.com/10827585/156508259-7c06f86a-a1f0-402b-a1db-78dd3ecc1e05.mov

@sojan-official This is working as per expectations, please have a look.

**Comment by sojan-official** [comment]
@tejaswinichile  @pranavrajs should we should show an indication in the message that it's a story mention. possibly adding a link to the original story like how we have for tweets?

**Comment by tejaswinichile** [comment]
@sojan-official @pranavrajs This would require adding one more call to fetch the link on which the content won't be available after 24 hrs, and this would only work for story mentions not for sharing post's image/video.

**Comment by tejaswinichile** [comment]
https://user-images.githubusercontent.com/10827585/156582863-48302fff-2c29-41f8-bc95-169c9d44ff8f.mov

Please check, I have added the message content for the story mentioned with the link to the story because Instagram provided the public CDN link to see the story, I have created the story link with the data that we are getting into the Facebook call.

File: app/builders/messages/messenger/message_builder.rb [hunk 1]
```diff
@@ -39,4 +43,30 @@ def file_type_params(attachment)
...
+      @inbox.channel.authorization_error!
+      raise
+    rescue StandardError => e
+      result = {}
+      Sentry.capture_exception(e)
+    end
+    story_id = result['story']['mention']['id']
+    story_sender = result['from']['username']
+    link_to_the_stroy = "https://www.instagram.com/stories/#{story_sender}/#{story_id}"
+    message.content = "#{story_sender} mentioned you in the story: #{link_to_the_stroy}"
```
Comment (pranavrajs) [inline app/builders/messages/messenger/message_builder.rb:69]: @tejaswinichile Please i18n constant here.

**Comment by sojan-official** [comment]
@tejaswinichile  rather than rendering the link inside the message, can we do it like how we do it for tweets? 

Screenshot 2022-03-04 at 7 24 22 PM

**Comment by ramphy** [comment]
Tested and working. Thank you @tejaswinichile

**Review by pranavrajs — APPROVED** [review]

**Pull request merged**